### PR TITLE
Build with dune

### PR DESCRIPTION
**Pull request by ghuysmans**

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
Hi, this was a temporary fork of https://github.com/sagotch/ocaml-gpx for a group school project.
I don’t really remember why we hadn’t ported our changes upstream but could you try to open a PR there instead?

If you really need one of our changes i think you can also make a separate PR there cherry-picking one or sereval of:
* https://github.com/LamaUrbain/ocaml-gpx/commit/d7b7cd491da3d8db953d9d4597d97268f396b8c4
* https://github.com/LamaUrbain/ocaml-gpx/commit/386a40bcd6af48c93b039b687179cdceb57aaad1
* https://github.com/LamaUrbain/ocaml-gpx/commit/67614ca1023e9bfb164a4ea2e1102e3a5e1d788a
* https://github.com/LamaUrbain/ocaml-gpx/commit/bcdb13691fceb9982bb895f8ed302e72ab7ceb0f

File: dune-project [hunk 1]
```diff
@@ -0,0 +1,26 @@
+(lang dune 2.0)
+
+(name gpx)
+(version 1.1.1)
+
+(generate_opam_files true)
```
Comment (kit-ty-kate) [inline dune-project:6]: Also it seems like you missed adding the generated `gpx.opam` file to the PR. Without it `opam pin` would not work properly.

As a side note as well, i personally don’t think generating the opam file from dune should be done. It make it harder to maintain and is riddled with bugs.

**Comment by ghuysmans** [comment]
Hi,
Thanks for your advice!

File: dune-project (hunk 1 above)
Comment (ghuysmans) [reply dune-project:6]: Oh, I should have tested it... thanks again!

**Pull request closed without merging**